src/modules/core/components/ColorSelect/ColorSelect.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useCallback, useMemo, ReactNode, ComponentProps } from 'react';
+import { defineMessages } from 'react-intl';
+
+import { Select, SelectOption, Form } from '~core/Fields';
+import ColorTag, { Color } from '~core/ColorTag';
+
+import styles from './ColorSelect.css';
+
+const MSG = defineMessages({
+  labelColorSelect: {
+    id: 'ColorSelect.labelColorSelect',
+    defaultMessage: 'Select color',
+  },
+});
+
+interface FormValues {
+  activeColor: Color;
+}
+
+interface Props {
+  /** Should `select` be disabled */
+  disabled?: boolean;
+
+  /** Active color */
+  activeOption: Color;
+
+  /** Callback function, called after value is changed */
+  onColorChange?: (color: Color) => any;
+}
+
+const displayName = 'ColorSelect';
+
+const ColorSelect = ({ disabled, activeOption, onColorChange }: Props) => {
+  const onChange = useCallback(
+    (color: Color) => {
+      if (onColorChange) {
+        return onColorChange(color);
+      }
+      return null;
+    },
+    [onColorChange],
+  );
+
+  const renderActiveOption = useCallback<
+    (option: SelectOption | undefined, label: string) => ReactNode
+  >(() => {
+    return <ColorTag color={activeOption || Color.Yellow} />;
```

Comment:
Please make sure the active color is vertically centered with the caret icon. 

Currently it's not:

![Screenshot from 2020-12-09 20-22-43](https://user-images.githubusercontent.com/1193222/101671737-8a348780-3a5d-11eb-9754-17be85f8afa1.png)